### PR TITLE
Add Excel DateTime parsing option in File Importer

### PR DESCRIPTION
**Pull request by Auriga2**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
The `EXCEL` is added as an option for parsing native excel datetime into Constellation which is available in the  `DateTime` `Formatter` in the `File Importer`.

### Verification Process

1. Create an excel spreadsheet with a column with datetimes represented as a numeric value
2. Open File Importer in Constellation and load file in
3. Try to parse the intended column as a datetime
4. Right click and set the `DateTime` attribute format to `Datetime`->`EXCEL` and now parse the column as a datetime

### Applicable Issues
https://github.com/constellation-app/constellation/issues/1663

### CODE REVIEW

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/translator/DatetimeAttributeTranslator.java [hunk 1]
```diff
@@ -112,6 +115,10 @@ public String translate(final String value, final PluginParameters parameters) {
                     dateTime = df.parse(value);
                     break;
                 }
+                case EXCEL: {
+                    dateTime = TemporalFormatting.zonedDateTimeFromLong(DateUtil.getJavaDate(Double.parseDouble(value)).getTime());
+                    break;
+                }
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/translator/DatetimeAttributeTranslator.java:121]: I know it's there in a couple other cases but if you're able to remove the curly braces (they're not needed), that would be great

**Review by antares1470 — CHANGES_REQUESTED** [review]
I don't think the conversion is working quite right. I passed in a file with a datetime value of 1.00 which should be 1/1/1900 00:00:00 but when I applied the excel formatter it became 31/12/1899 14:00:00. A similar margin of error was occurring for other datetimes as well.

A couple of other points:
1) Are you able to order the formatters so that Excel goes before Custom rather than after? I think Custom makes sense to have as the last option
2) If you're able to add a whatsnew entry, that would be great

**Review by arcturus2 — APPROVED** [review]

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/translator/DatetimeAttributeTranslator.java (hunk 1 above)
Comment (Auriga2) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/translator/DatetimeAttributeTranslator.java:121]: @antares1470, When I debugged, the epoch time for above example is -2209023000000 
(https://www.epochconverter.com/) produces both days below, so looks like all dates are converted to GMT. Should we display it in local time zone?
image

**Comment by Auriga2** [comment]
Fixed other requested changes.

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/translator/DatetimeAttributeTranslator.java (hunk 1 above)
Comment (antares1470) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/translator/DatetimeAttributeTranslator.java]: Right I guess that makes sense. If converting the excel time to epoch still produces the same value in Constellation and then I suppose for now it makes sense to keep as is. I suppose from my perspective it just looks a little weird inputing a time into a spreadsheet and then it importing as a different time in Constellation. Maybe what I will do though is put in a new issue to give an option to specify a time zone when applying the formatter (excel or epoch)

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
 == 3030-12-31 Getting Started
 <p>If you're new to Constellation, read the <a href="" helpId="au.gov.asd.tac.constellation.functionality.gettingstarted">getting started guide</a>.</p>
 
+== 2022-03-22 New Datetime formatter in the File Importer
+<p>New Datetime formatter 'EXCEL' is added as an option for parsing native excel datetime into Constellation which is available as an 'Datetime Format' in the 'Datetime Parameters' dialog when the DateTime attribute is right clicked and 'Formatter' -> 'Datetime' is choosen.</p>
+
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt:6]: I think I would split up this sentence a little. Maybe something like "A new Datetime formatter has been added to the File Importer for parsing native Excel datetimes into Constellation. You can use it by right-clicking an attribute, navigating to Formatter -> Datetime and choosing 'EXCEL' from the format dropdown list"

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt (hunk 2 above)
Comment (Auriga2) [reply CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt]: Yes it' much better. Changed now.

**Review by antares1470 — APPROVED** [review]

**Comment by Auriga2** [comment]
Thanks @GammaVel for thorough testing. A further investigation revealed the problem.  `DateUtil.getJavaDate` I use in the conversion applies the user's time zone if a time zone is not specified, which made all the confusion. If I specify "GMT" there, it works similar to EPOCH converter. I'll fix that here and the rest below can be handled separately.

https://github.com/constellation-app/constellation/issues/1673 still applies after fixing this, because the last part of the converted datetime implies they are GMT, where it could actually be any time zone. (Added more details in that issue.)
image

**Comment by Auriga2** [comment]
Created https://github.com/constellation-app/constellation/pull/1685.

**Pull request merged**